buildSrc/src/main/groovy/com/thoughtworks/go/build/license/HtmlReportBuilder.groovy

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.build.license
+
+import com.github.jk1.license.*
+import groovy.json.JsonSlurper
+import groovy.xml.MarkupBuilder
+import org.gradle.api.Project
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import static com.thoughtworks.go.build.license.OpensourceOrgLicenseUrlGenerator.getLicenseURLs
+
+class HtmlReportBuilder {
+  private def javaDependencies
+  private final Project rootProject
+  private final AtomicInteger counter
+  private List<Object> javascriptDependencies
+  private File licenseReportDir
+
+  HtmlReportBuilder(Project rootProject) {
+    this.rootProject = rootProject
+    this.counter = new AtomicInteger()
+    this.javascriptDependencies = new JsonSlurper().parse(getClass().getResourceAsStream("/included_javascript_license_info.json"))
+    this.licenseReportDir = new File("$rootProject.projectDir.absolutePath/server/webapp/static/license_report")
```

Comment:
Since this is library code, ideally it should not touch the filesystem, unless specific files are passed in. It should not assume project and filesystem structure.